### ISSUE TITLE
Execution of an await for loop doesn't stop

### ISSUE THREAD

**Issue by gosoccerboy5** [dart-lang/dart-pad#1986]
## What happened?

Dartpad didn't stop the execution of an `await for` loop when I clicked the Run button again.

## Steps to reproduce problem

Go to https://dartpad.dev/?id=952e8d67f9b5693cf1c215745e588ad4&null_safety=true and click `Run` multiple times

### Browser

Browser: Chrome

Version: 91

Are you using any extensions/plugins that affect website behavior
(particularly those that affect iframes, such as ad blockers)?

no, I don't think so

Are there any warnings or errors in your browser's JavaScript console?
If so, paste them below:

none

### Machine

Operating system: MacOS

Version: 11.4

### Your code

What code was in the editor, if any, when the failure occurred? You
can paste it in below:

https://gist.github.com/gosoccerboy5/952e8d67f9b5693cf1c215745e588ad4

### DartPad's output

Did DartPad print anything to the console pane? If so, paste it below:

(just reproduce the bug to see yourself)

**Comment by domesticmouse** [comment]
Please see https://github.com/dart-lang/dart-pad/issues/1817 for a full discussion of this issue. 

Tl;dr: engineering trade off between speed and correctness.

**Comment by gosoccerboy5** [comment]
ok
